storage/core/src/main/java/com/hortonworks/streamline/storage/impl/jdbc/JdbcStorageManager.java

```diff
@@ -81,6 +84,14 @@ public void addOrUpdate(Storable storable) throws StorageException {
     }
 
     @Override
+    public void update(Storable storable) {
+        if (queryExecutor.update(storable) == 0) {
```

Comment:
The query executor is low-level and just returns the number of rows updated. This logic is common for mysql/postgres so would like to retain it here.

So assuming we passed a Storable with proper values (including PK and optional Version field), only way update would return 0, is if that row (with that PK plus optional version field) was not found which means there was a concurrent update or delete for that same record. If there was some other DB issues, it would manifest as some JDBCException which will propagate up.